packages/flutter_tools/lib/src/project.dart

```diff
@@ -153,7 +153,7 @@ class IosProject {
   /// The parent of this project.
   final FlutterProject parent;
 
-  static final RegExp _productBundleIdPattern = RegExp(r'^\s*PRODUCT_BUNDLE_IDENTIFIER\s*=\s*(.*);\s*$');
+  static final RegExp _productBundleIdPattern = RegExp(r'^\s*PRODUCT_BUNDLE_IDENTIFIER\s*=\s*["]?(.*?)["]?;\s*$');
```

Comment:
why `["]?` instead of `"?` ?

Reply:
as written this will match `PRODUCT_BUNDLE_IDENTIFIER=FOO:"BAR";` as `FOO:"BAR`, and will match `PRODUCT_BUNDLE_IDENTIFIER="FOO;` (even though the end quote is missing). Is that intentional?

Reply:
The [] brackets are not required, but they don't hurt. I am not a fan of just single characters within regex, that's why I use the set. This is just a personal preference

BundleIdentifier has the following limitation from Apple:
Maximum Length: 155
Pattern: ^[A-Za-z0-9\.\-]+$

It is therefore not possible, to have a " sign within an identifier.

In some xcode versions, the identifier will be stored like:
PRODUCT_BUNDLE_IDENTIFIER="com.github.myAwesome.app";
In other xcode versions, it will be stored as:
PRODUCT_BUNDLE_IDENTIFIER=com.github.myAwesome.app;